src/lib/Movement/moveTo.ts

```diff
@@ -92,6 +92,77 @@ export const moveTo = (
     clearCachedPath(creep, cache);
   }
 
+  // Dynamic choose weight for roads, plains and swamps depending on body.
+  // Skip if power creep or any costs has been manually set.
+  if (DEBUG) logCpu('getting default plain and swamp costs');
+  const manuallyDefinedCosts = opts?.roadCost || opts?.plainCost || opts?.swampCost;
+  if ('body' in creep && !manuallyDefinedCosts) {
```

Comment:
Let's separate this entire calculation to a new function:

```
function defaultTerrainCosts(creep: Creep): Required<Pick<MoveOpts, "roadCost" | "plainCost" | "swampCost">>
```

This will make it easier to read & test the functionality separately. Then we can do:

```
if ('body' in creep && !manuallyDefinedCosts) {
   actualOpts = { ...actualOpts, ...defaultTerrainCosts(creep) }
}
```